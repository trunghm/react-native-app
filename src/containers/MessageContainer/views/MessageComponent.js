import React, {Component} from 'react'
import {View, Text} from 'react-native'
import styles from './styles';
import I18n from '../../../I18n'

export default class MessageComponent extends Component {
  constructor(props) {
    super(props)
  }

  render() {
    return (
      <View style={styles().container}>
        <Text style={styles().text}>{I18n.t('message_page.title_page')} </Text>
      </View>
    )
  }
}

