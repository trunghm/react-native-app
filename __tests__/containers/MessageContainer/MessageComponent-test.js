/**
 * @format
 */

import 'react-native';
import React from 'react';
import MessageComponent from '../../../src/containers/MessageContainer/views';

// Note: test renderer must be required after react-native.
import {mount, shallow} from 'enzyme';
import renderer from  "react-test-renderer";
import EStyleSheet from 'react-native-extended-stylesheet';


const originalError = console.error;

beforeAll(() => {
  console.error = jest.fn();
  EStyleSheet.build({
  });
});

afterAll(() => {
  console.error = originalError;
});

describe('Testing  MessageComponent component', () => {

  it('renders snapshot as expected', () => {
    const wrapper = renderer.create(
      <MessageComponent />
    );

    expect(wrapper).toMatchSnapshot();

  });

  it('renders as expected', () => {
    const wrapper = mount(
      <MessageComponent />
    );

    expect(wrapper).toMatchSnapshot();

  });

});
