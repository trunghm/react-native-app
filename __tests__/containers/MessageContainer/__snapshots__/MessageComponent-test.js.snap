// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing  MessageComponent component renders as expected 1`] = `ReactWrapper {}`;

exports[`Testing  MessageComponent component renders snapshot as expected 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "flex": 1,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#333333",
      }
    }
  >
    message_page.title_page
     
  </Text>
</View>
`;
